Regex Atoms

jared-hughes opened the pull request:
- `œF`: Regex Findall: Find all instances of regex `x` in string `y`; return a list of matches
 - `œḞ`: Regex Finditer: Find all instances of regex `x` in string `y`; return a list of lists of groups
 - `œṠ`: Regex Sub: Take arguments substitution configuration `[x1,x2]` and y. Substitute all instances of regex `x1` with replacement string `x2` in string `y`.
 - `œR`: Regex Search: Give a boolean denoting if regex `x` matches string `y` in any position.
 - `œṢ`: Regex Split: Split string `y` at occurrences of regex `x`.
 - `ÐR`: Regex sub quick: `<regex><sub>ÐR` replaces all occurrences of `<regex>` with `<sub>` applied to each occurrence’s list of groups and span

Some possible issues: 

 - If `œṠ` is given a left argument/configuration which is a string, it substitutes all instances of the first character of the string with the second character of the string in the right argument, which is the same as atom `y`.
 - `ÐR` uses `to_string(x) = ''.join(x) if is_string(x) else str(x)` to cast the output of `<sub>` to string. Not sure if there is a better casting method.

jared-hughes committed:
Add jelly regex

jared-hughes committed:
Minor clean up

GolfingSuccess commented:
I suggest making `œṠ` to act more like`y`, i.e. `[x1, x2, x3, x4]` would substitute `x1` with `x2` and `x3` with `x4`, with earlier matches having priority.

jared-hughes committed:
Modified Vectorization on œṠ and ÐR

- œṠ works like link y now
- ÐR applies all regexps in link[0] in order if link[0] is a list of strings/regexps

DennisMitchell commented:
A few issues:

1. I'm currently unable to test this because line 1 of jelly.py changed both on master and here. I *think* that could be resolved by re-adding `parser`, but I'll see if I can resolve the conflicts in another way.

2. Many of new the atoms returns `str`. Jelly doesn't have a string type, so this is bound to cause issues. A single `str` can be cast to `list`.

3. `œR` returns `bool`, which is another type Jelly doesn't have.
